internal/ethapi/dposparse.go

```
@@ -0,0 +1,190 @@
+		return types.AddCandidateTxData{}, err
+	}
+
+	// parse deposit
+	if data.Deposit, err = unit.ParseCurrency(depositStr); err != nil {
+		return types.AddCandidateTxData{}, err
+	}
+
+	return
+}
+
+// parseCandidates will convert the string to a list of candidate address
+func parseCandidates(candidates string) ([]common.Address, error) {
+	// strip all white spaces
+	candidates = strings.Replace(candidates, " ", "", -1)
+
+	// convert it to a list of candidates
+	candidatesList := strings.Split(candidates, ",")
+
+	// candidates addresses conversion
+	var candidateAddresses []common.Address
+	for _, candidate := range candidatesList {
+		addr := common.HexToAddress(candidate)
+		candidateAddresses = append(candidateAddresses, addr)
+	}
+
+	return candidateAddresses, nil
+}
+
+// parseRewardRatio is used to convert the ratio from string to uint64
+// and to validate the parsed ratio
+func parseRewardRatio(ratio string) (uint64, error) {
+	// convert the string to uint64
+	rewardRatio, err := unit.ParseUint64(ratio, 1, "")
+	if err != nil {
+		return 0, ErrParseStringToUint
+	}
+
+	// validate the rewardRatio and return
+	return rewardRatioValidation(rewardRatio)
+}
+
+// rewardRatioValidation is used to validate the reward ratio
+func rewardRatioValidation(ratio uint64) (uint64, error) {
+	// check if the reward ratio
+	if ratio > dpos.RewardRatioDenominator {
+		return 0, ErrInvalidAwardDistributionRatio
+	}
+	return ratio, nil
+}
+
+// defaultAccount will return the first account address from the first wallet
+func defaultAccount(account *accounts.Manager) common.Address {
+	// check if there are existing account, if so, use the first one as
+	// default account
+	if wallets := account.Wallets(); len(wallets) > 0 {
+		if walletAccounts := wallets[0].Accounts(); len(walletAccounts) > 0 {
+			return walletAccounts[0].Address
+		}
+	}
+
+	// otherwise, return empty account address
+	return common.Address{}
```

Would it be better to return an error and tell user no account is available?

---

agree, fixed